Tests/Common/PasswordManipulationTest.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Framework\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Zest\Common\PasswordManipulation;
+
+class PasswordManipulationTest extends TestCase
+{
+    public function testGeneratePassword()
+    {
+        $pass = new PasswordManipulation();
+        $this->assertNotEmpty($pass->generatePassword(11));
+    }
+
+    public function testCountTes()
+    {
+        $pass = new PasswordManipulation();
+        $this->assertSame(6, $pass->len(123456));
+    }
+
+    public function testIsValid()
+    {
+        $pass = new PasswordManipulation();
+        $this->assertTrue($pass->isValid($pass->generatePassword(15)));
```

Comment:
It looks like the random password doesn't always have correct validation.

The failed assertion may happen:

```
1) Framework\Tests\PasswordManipulationTest::testIsValid

Failed asserting that false is true.

/home/travis/build/zestframework/Zest_Framework/Tests/Common/PasswordManipulationTest.php:25
```

Reply:
I have to say that the `isValid` method is problematic.

It should be concerned and enhanced.

Reply:
Lets consider this code

```php
       $password = new PasswordManipulation();
       $pass = $password->generatePassword(5);
       var_dump($pass);
       var_dump($password->isValid($pass));
```

For now forget unit Test, run this with Zest framework it will produce right result, what the probem with Unit test, idk

Reply:
No. That's not correct.

The password is generated randomly.

That is, the unit test will be failed probably.

Nobody can guarantee that the unit test will be failed.

Don't forget that the unit tests can help us to "find errors".

And it does, it helps us to find `isValid` method approach has been problematic.

Reply:
I think `generatePassword` is problematic that cant generate strong password , right?

Reply:
If `isValid` can help us to validate random password correctly, the `generatePassword` will be problematic probably.

Reply:
@peter279k  see all tests passed now, it mean problem with `generatePassword()` method